### PR TITLE
SNOW-802306 improve tests

### PR DESCRIPTION
**Pull request by sfc-gh-mkeller**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Helps with SNOW-802306
2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   In this PR I should have made some tests more resilient against name collision and provide more information when they fail.

### CODE REVIEW

File: tests/integ/test_pandas_to_df.py [hunk 1]
```diff
@@ -149,22 +153,22 @@ def test_write_pandas(session, tmp_table_basic):
         columns=["id".upper(), "foot_size".upper(), "shoe_model".upper()],
     )
 
-    df = session.write_pandas(pd, tmp_table_basic)
+    df = session.write_pandas(pd, tmp_table_basic, overwrite=True)
     results = df.to_pandas()
     assert_frame_equal(results, pd, check_dtype=False)
 
     # Auto create a new table
-    session._run_query('drop table if exists "tmp_table_basic"')
```
Comment (sfc-gh-sfan) [inline tests/integ/test_pandas_to_df.py:157]: Wow this does not seem expected at all. Thanks for catching it.

**Review by sfc-gh-sfan — COMMENTED** [review]
Can you help me understand the changes in the other two tests that are not `test_pandas_to_df`?

**Comment by sfc-gh-mkeller** [comment]
> Can you help me understand the changes in the other two tests that are not `test_pandas_to_df`?

Essentially both tests used hardcoded table names, so I really think that different testruns stepped on each other's toes. So for the one tests that meant to call the random name function. While for the other we already made a random name, but we never used it, it simply looked like someone forgot to add the f-strings.

**Comment by sfc-gh-mkeller** [comment]
NVM, I misread what you said. You meant he other 2 tests.

I parametrized the SQLs. Here are the [pytest docs](https://docs.pytest.org/en/6.2.x/parametrize.html).
Take a look at how the failure shows [here](https://ci-dev-142.int.snowflakecomputing.com/job/SnowparkPythonSnowflakePythonClientRegressRunner/192/console)
```2023-05-01 15:06:09 __________________________ test_show_database_objects __________________________
2023-05-01 15:06:09 
2023-05-01 15:06:09 session = <snowflake.snowpark.session.Session object at 0x7fb34e86df10>
2023-05-01 15:06:09 
2023-05-01 15:06:09     @pytest.mark.skipif(IS_IN_STORED_PROC_LOCALFS, reason="Large result")
2023-05-01 15:06:09     def test_show_database_objects(session):
2023-05-01 15:06:09         sqls = [
2023-05-01 15:06:09             "show schemas",
2023-05-01 15:06:09             "show objects",
2023-05-01 15:06:09             "show external tables",
2023-05-01 15:06:09             "show views",
2023-05-01 15:06:09             "show columns",
2023-05-01 15:06:09             "show file formats",
2023-05-01 15:06:09             "show sequences",
2023-05-01 15:06:09             "show stages",
2023-05-01 15:06:09             "show pipes",
2023-05-01 15:06:09             "show streams",
2023-05-01 15:06:09             "show tasks",
2023-05-01 15:06:09             "show user functions",
2023-05-01 15:06:09             "show external functions",
2023-05-01 15:06:09             "show procedures",
2023-05-01 15:06:09             "show tables",
2023-05-01 15:06:09             "show parameters",
2023-05-01 15:06:09             "show shares",
2023-05-01 15:06:09             "show warehouses",
2023-05-01 15:06:09             "show transactions",
2023-05-01 15:06:09             "show locks",
2023-05-01 15:06:09             "show regions",
2023-05-01 15:06:09         ]
2023-05-01 15:06:09         for sql in sqls:
2023-05-01 15:06:09 >           Utils.verify_schema(sql, session.sql(sql).schema, session)
2023-05-01 15:06:09 
2023-05-01 15:06:09 tests/integ/scala/test_result_schema_suite.py:74: 
2023-05-01 15:06:09 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2023-05-01 15:06:09 
2023-05-01 15:06:09 sql = 'show parameters'
2023-05-01 15:06:09 expected_schema = StructType([StructField('"key"', StringType(), nullable=True), StructField('"value"', StringType(), nullable=True), St...eld('"activate_version"', StringType(), nullable=True), StructField('"partial_rollout"', StringType(), nullable=True)])
2023-05-01 15:06:09 session = <snowflake.snowpark.session.Session object at 0x7fb34e86df10>
2023-05-01 15:06:09 
2023-05-01 15:06:09     @staticmethod
2023-05-01 15:06:09     def verify_schema(sql: str, expected_schema: StructType, session: Session) -> None:
2023-05-01 15:06:09         session._run_query(sql)
2023-05-01 15:06:09         result_meta = session._conn._cursor.description
2023-05-01 15:06:09     
2023-05-01 15:06:09 >       assert len(result_meta) == len(expected_schema.fields)
2023-05-01 15:06:09 E       AssertionError
2023-05-01 15:06:09 
2023-05-01 15:06:09 tests/utils.py:233: AssertionError
```
I find this a little hard to see wich SQL caused the error.

With this we essentially run the test many times, so it becomes much more obvious which SQL failed. This technically doesn't make things any better, but instead of `test_describe_schema_matches_execute_schema_for_show_queries` failing it'd be `test_describe_schema_matches_execute_schema_for_show_queries[show parameters]`.

I'm working with the author of this change to actually resolve this issue

**Comment by sfc-gh-stan** [comment]
In `Utils.verify_schema`, you could also add an error message like `assert <x1> == <x2>, f"contents of x1: {x1}"` for more information.

**Review by sfc-gh-sfan — APPROVED** [review]

**Pull request merged**